### PR TITLE
Remove dependent on session auth for api/v1 routers

### PR DESCRIPTION
**Pull request by lunny**
partially replace #16052

### CODE REVIEW

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
@lunny `TestSearchIssuesWithLabels` -> do not work

File: routers/api/v1/api.go [hunk 1]
```diff
@@ -596,11 +594,9 @@ func buildAuthGroup() *auth.Group {
...
 // Routes registers all v1 APIs routes to web application.
-func Routes(sessioner func(http.Handler) http.Handler) *web.Route {
+func Routes() *web.Route {
 	m := web.NewRoute()
 
-	m.Use(sessioner)
-
 	m.Use(securityHeaders())
 	if setting.CORSConfig.Enabled {
 		m.Use(cors.Handler(cors.Options{
```
Comment (6543) [inline routers/api/v1/api.go:602]: in L608:
```diff
-			AllowedHeaders:   []string{"Authorization", "X-CSRFToken", "X-Gitea-OTP"},
+			AllowedHeaders:   []string{"Authorization", "X-Gitea-OTP"},
```

**Review by 6543 — APPROVED** [review]

**Comment by wxiaoguang** [comment]
There are still calls to `/api/v1` from UI after 
* https://github.com/go-gitea/gitea/pull/19318


```
Powered by Gitea Version: 1.17.0+dev-409-g783a02188 Page: 85ms Template : 15ms 
POST https://try.gitea.io/api/v1/repos/wxiaoguang/test/markdown
```

**Comment by lunny** [comment]
> markdown

Good catch, I will fix them in this PR.

**Comment by lunny** [comment]
> There are still calls to `/api/v1` from UI after
> 
> * [Never use /api/v1 from Gitea UI Pages #19318](https://github.com/go-gitea/gitea/pull/19318)
> 
> ```
> Powered by Gitea Version: 1.17.0+dev-409-g783a02188 Page: 85ms Template : 15ms 
> POST https://try.gitea.io/api/v1/repos/wxiaoguang/test/markdown
> ```

Done.

**Review by 6543 — APPROVED** [review]
ah my grep missed the "APIURL" string in templates ... thanks for catch them too

**Comment by wxiaoguang** [comment]
More information: using `HTMLURL` in templates is the root case that Gitea can only serve web under ROOT_URL. That's why we have to force users to set ROOT_URL correctly, and that's that PR comes:

* https://github.com/go-gitea/gitea/pull/19291

If all URLs in template can be refactored to use Realtive URL correctly, then we do not need to force users set ROOT_URL correctly again and again.

**Review by wxiaoguang — APPROVED** [review]

**Comment by 6543** [comment]
@lunny we might add some linter to ensure that `templates` and `web_src` dont contain any `api/v1` or `APIURL` ?

**Pull request merged**